### ISSUE TITLE
Better Solution for node-gamepad & node-webkit Compatibility

### ISSUE THREAD

**Issue by gluxon** [gluxon/DriverStation.js#14]
As outlined in the [3rd party modules with C/C++ addons](https://github.com/rogerwang/node-webkit/wiki/Using-Node-modules#3rd-party-modules-with-cc-addons) section of the node-webkit wiki, native modules with node-webkit run with a different ABI (application binary interface) than standard node. @gixxy came up with a nice solution in pull request #13, but it would be nice to automate this. (Say with Grunt)

**Comment by gixxy** [comment]
Agreed 100%. The amount of manual intervention required to get node-gamepad compiled with node-pre-gyp was ridiculous, and is certainly not automated. I know of Grunt.js but I will definitely have to look into using it. I did make sure to put all my documentation on the task in the README.

**Comment by gluxon** [comment]
The documentation you added was extremely helpful, thank you for that. I mentioned Grunt as a possible solution, not sure if it's the best yet.

**Comment by ExploitSage** [comment]
Another thing I have since noticed is that Node-webkit v0.11 is based on Chromium 38, which supports the Gamepad API. It is a potential integrated solution considering the node-gamepad library is not currently compatible with Node-Webkit v0.11

**Comment by gluxon** [comment]
The reason I didn't love the Gamepad API was because of lack of support for the Logitech Extreme 3D Pro. It appears [that issue](https://code.google.com/p/chromium/issues/detail?id=110013) has been fixed on Windows now.

I'm tempted to a create a gamepad-api branch and start the refactor there given the node-gamepad headaches. OS X and Linux may need their own solutions until Chrome finishes their implementation though. @gixxy, what do you think?

**Comment by ExploitSage** [comment]
I figure it is at least worth trying out. Branch and try it. I am less available now due to other commitments than I was the last 4 weeks, (between college, work(s), and mentoring a team) but I will help when and where I can. My code should be fairly clean and understandable. I will always respond to any questions you need answered though.

Even if the gamepad-api isn't quite up to snuff cross-platform wise yet, if we have it generally ported, and ready to go when it is, we will be a step ahead.

**Comment by bb010g** [comment]
bumpity bump
